modules/images/webp-uploads/load.php

```diff
@@ -157,6 +158,41 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 
 add_filter( 'wp_generate_attachment_metadata', 'webp_uploads_create_sources_property', 10, 2 );
 
+/**
+ * Filter the image editor default output format mapping to select the most appropriate
+ * output format depending on desired output formats and supported mime types by the image
+ * editor.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $output_format The image editor default output format mapping.
+ * @param string $filename      Path to the image.
+ * @param string $mime_type     The source image mime type.
+ * @return string The new output format mapping.
+ */
+function webp_uploads_filter_image_editor_output_format( $output_format, $filename, $mime_type ) {
+	// Use the original mime type if this type is allowed.
+	$valid_mime_transforms = webp_uploads_get_upload_image_mime_transforms();
+	if (
+		! isset( $valid_mime_transforms[ $mime_type ] ) ||
+		in_array( $mime_type, $valid_mime_transforms[ $mime_type ], true )
+	) {
+		return $output_format;
+	}
+
+	// Find the first supported mime type by the image editor to use it as the default one.
+	foreach ( $valid_mime_transforms[ $mime_type ] as $target_mime ) {
+		if ( wp_image_editor_supports( array( 'mime_type' => $target_mime ) ) ) {
+			$output_format[ $mime_type ] = $target_mime;
+			break;
+		}
+	}
+
+	return $output_format;
+}
+
+add_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format', 10, 3 );
```

Comment:
Could be considered a personal preference, but I think it's more common with functional WP plugins to have the `add_filter` / `add_action` call right below the callback function.
```suggestion
}
add_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format', 10, 3 );
```

Reply:
Updated.